gradio/chat_interface.py

```diff
@@ -618,13 +619,16 @@ async def _submit_fn(
             response = await anyio.to_thread.run_sync(
                 self.fn, *inputs, limiter=self.limiter
             )
-        if isinstance(response, tuple):
+        if self.additional_outputs:
             response, *additional_outputs = response
         else:
             additional_outputs = None
         history = self._append_message_to_history(message, history, "user")
-        response_ = self.response_as_dict(response)
-        history = self._append_message_to_history(response_, history, "assistant")  # type: ignore
+        response_ = [response] if not isinstance(response, list) else response
```

Comment:
We should add this logic to `_stream_fn` to support streaming. Otherwise LGTM!

Reply:
Nice catch thank you!